atomic_reactor/util.py

```diff
@@ -738,6 +738,46 @@ def __init__(self, registry, insecure=False, dockercfg_path=None, access=None):
 
         self.session = get_retrying_requests_session()
 
+    @classmethod
+    def create_from_config(cls, source_registry, pull_registries, registry=None, access=None):
+        """
+        Create a session for the specified registry based on configuration in reactor config map.
+
+        If the registry is configured in pull_registries, use that configuration. Otherwise,
+        use the default source_registry configuration (if present, else an error is thrown).
+
+        :param source_registry: source_registry configuration from reactor config map (or None)
+                                See get_source_registry() in reactor_config plugin
+        :param pull_registries: pull_registries configuration from reactor config map
+                                See get_pull_registries() in reactor_config plugin
+        :param registry: str, registry to create session for (as hostname[:port], no http(s))
+                         If not specified, the default source_registry will be used
+        :param access: List of actions this session is allowed to perform, e.g. ('push', 'pull')
+
+        :return: RegistrySession for the specified registry
+        """
+        if registry is None:
+            if source_registry is not None:
+                matched_registry = source_registry
+            else:
+                msg = "No source_registry configured, cannot create default session"
+                raise RuntimeError(msg)
+        else:
+            # Try to match a registry in pull_registries, if no match found, use source_registry
+            matched_registry = next(
```

Comment:
@chmeliik; Why `next()`? This skips the first iteration, right?

Reply:
`next()` returns the, well, *next* element in an iterator, e.g.:
```python
>>> it = iter([1, 2, 3])
>>> next(it)
1
>>> next(it)
2
>>> next(it)
3
>>> next(it)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```
I'm not sure how idiomatic my code is, but this is what it does:
```python
# 1. Make iterator of all registries that match the registry given as a parameter
# (could be empty)
it = (reg for reg in pull_registries if reg['uri'].docker_uri == registry)

# 2. Take first element from this iterator, or the default if iterator is empty
matched_registry = next(it, source_registry)
```

Reply:
Got it. Thanks!